src/frontend/js/components/OpenEdxFullNameForm/index.tsx

```diff
@@ -0,0 +1,134 @@
+import { Button, Input } from '@openfun/cunningham-react';
+import { FormattedMessage, defineMessages, useIntl } from 'react-intl';
+import { FormProvider, useForm } from 'react-hook-form';
+import * as Yup from 'yup';
+import { yupResolver } from '@hookform/resolvers/yup';
+import { useEffect, useState } from 'react';
+import { useSession } from 'contexts/SessionContext';
+import useOpenEdxProfile from 'hooks/useOpenEdxProfile';
+import Form, { getLocalizedCunninghamErrorProp } from 'components/Form';
+import { Spinner } from 'components/Spinner';
+import Banner, { BannerType } from 'components/Banner';
+
+const messages = defineMessages({
+  fullNameInputLabel: {
+    id: 'components.OpenEdxFullNameForm.fullNameInputLabel',
+    description: 'Label of "fullName" field openEdx full name form',
+    defaultMessage: 'FullName',
+  },
+  submitButtonLabel: {
+    id: 'components.OpenEdxFullNameForm.submitButtonLabel',
+    description: 'Label of the submit button on openEdx full name form',
+    defaultMessage: 'Submit',
+  },
+  loadingLabel: {
+    id: 'components.OpenEdxFullNameForm.loadingLabel',
+    description: 'Label of the loading profile spinner on openEdx full name form',
+    defaultMessage: 'Loading...',
+  },
+  error: {
+    id: 'components.OpenEdxFullNameForm.loadingLabel',
+    description: 'Label of the loading profile spinner on openEdx full name form',
+    defaultMessage: 'Loading...',
+  },
+});
+
+export interface OpenEdxFullNameFormValues {
+  name: string;
+}
+
+const validationSchema = Yup.object().shape({
+  name: Yup.string().required(),
+});
+
+const OpenEdxFullNameForm = () => {
+  const intl = useIntl();
+  const { user } = useSession();
+  const [isSuccess, setIsSuccess] = useState<boolean>(false);
+  const {
+    data: openEdxProfileData,
+    methods: { update },
+    states: { isFetched, isPending },
+    error,
+  } = useOpenEdxProfile({
+    username: user!.username,
+    onUpdateSuccess: () => {
+      setIsSuccess(true);
+    },
+  });
+
+  const defaultValues = {
+    name: openEdxProfileData ? openEdxProfileData.name : '',
+  } as OpenEdxFullNameFormValues;
+  const form = useForm<OpenEdxFullNameFormValues>({
+    defaultValues,
+    mode: 'onBlur',
+    reValidateMode: 'onChange',
+    resolver: yupResolver(validationSchema),
+  });
+  const { register, handleSubmit, reset, formState } = form;
+
+  const onSubmit = (values: OpenEdxFullNameFormValues) => {
+    update(values);
+  };
+
+  useEffect(() => {
+    if (openEdxProfileData) {
+      reset({ name: openEdxProfileData.name });
+    }
+  }, [openEdxProfileData]);
+
+  if (!isFetched) {
+    if (isPending) {
+      return (
+        <Spinner>
+          <FormattedMessage {...messages.loadingLabel} />
+        </Spinner>
+      );
+    }
+
+    if (error) {
+      // display get error message
+      return <Banner type={BannerType.ERROR} message={error} />;
+    }
+  }
```

Comment:
I do not understand why you are checking if data are not yet fetched ?

Reply:
I made a mistake for the error that should be outside the `isFetched` condition.
Otherwise, the spinner need to be displayed when the query is pending for the first time and not when it's pending after submiting the form.